plugins/completion/cmp/sources/cmp-ai.nix

```diff
@@ -0,0 +1,91 @@
+{
+  lib,
+  helpers,
+  config,
+  ...
+}:
+with lib;
+let
+  cfg = config.plugins.cmp-ai;
+in
+{
+  meta.maintainers = [ maintainers.GaetanLepage ];
+
+  options.plugins.cmp-ai = {
+    settings = helpers.mkSettingsOption {
+      description = "Options provided to the `require('cmp_ai.config'):setup` function.";
+
+      options = {
+        max_lines = helpers.defaultNullOpts.mkUnsignedInt 50 ''
+          How many lines of buffer context to use.
+        '';
+
+        run_on_every_keystroke = helpers.defaultNullOpts.mkBool true ''
+          Generate new completion items on every keystroke.
+        '';
+
+        provider = helpers.defaultNullOpts.mkStr "HF" ''
+          Which AI provider to use.
+        '';
+
+        provider_options = helpers.defaultNullOpts.mkAttrsOf types.anything { } ''
+          Options to forward to the provider.
```

Comment:
I think we should link the README here, since they document available options for most providers.

Reply:
Yes, done.